fix(openai): streaming tool_call + logging multiple tool_call

guillaumefontan opened the pull request:
Fixes https://github.com/traceloop/openllmetry-js/issues/464 and https://github.com/traceloop/openllmetry-js/issues/465

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/traceloop/openllmetry-js?pullRequest=463) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/traceloop/openllmetry-js?pullRequest=463) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/traceloop/openllmetry-js?pullRequest=463) it.</sub>

nirga commented on packages/instrumentation-openai/src/instrumentation.ts:
```diff
@@ -388,27 +388,54 @@ export class OpenAIInstrumentation extends InstrumentationBase {
             }
           }
         }
+        // if (chunk.choices[0]?.delta.tool_calls) {
```
nit: remove?

guillaumefontan replied:
All done thanks!

nirga reviewed (commented):
Thanks! Left a small nit comment

guillaumefontan commented:
@nirga just a note, this will break anyone's code which uses
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.name`]` (will return undefined)
as the new format will become
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.0.name`]`.
To avoid any breaking changes, we could make it so that the first tool call remains
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.name`]`
while all other tool calls are indexed:
`span.attributes[`${SpanAttributes.LLM_COMPLETIONS}.0.function_call.1.name`]`.
However this may be confusing.

nirga commented:
@guillaumefontan I think we should match the way we've implemented this on Python - 
https://github.com/traceloop/openllmetry/blob/a1b82d8b272ce3861d88f5a2898d3170aa9a6df0/packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/chat_wrappers.py#L448

So it should be `tool_calls` (plural)

Also note the CI is broken on your PR

guillaumefontan commented:
Hi @nirga I've updated to tool_calls and fixed the lint warnings. However the build error:
`Run pip3 install chromadb
error: externally-managed-environment`
does not appear to be related to my changes. Let me know if there is anything I can do.

nirga reviewed (approved):
Sorry for the delay here, working on fixing the CI now. LGTM except for that!
